typing/typecore.ml

```diff
@@ -896,6 +896,18 @@ let check_recordpat_labels loc lbl_pat_list closed =
         end
       end
 
+(* Missed punning opportunity *)
+
+let check_missed_punning_opportunity f l =
+  if Warnings.is_active (Warnings.Missed_punning_opportunity "") then
+    List.iter
+      (fun x ->
+        match f x with
+        | Some (lab, loc, x) when lab.txt = x && lab.loc <> loc ->
+            Location.prerr_warning loc (Warnings.Missed_punning_opportunity x)
+        | _ -> ()
+      ) l
```

Comment:
Isn't this the same as a first-order function taking a list as argument, pre-composed with `List.filter_map`? If so, then I would personally have a preference for that simpler interface. (Or using `Seq` instead of `List` if we want to avoid creating intermediary structures, but I don't think that matters.)